### PR TITLE
Update CI and cloud.gov config to build with CircleCI

### PR DESCRIPTION
**Pull request by LindsayYoung**
This changeset removes the old Travis CI config in favor of the new CircleCI config.

It also removes the .python-version file from the repository for the time being. We were never able to move forward on the pip-tools install, so this is not needed at the moment and makes dealing with Python updates locally a bit more cumbersome at the moment. We can bring this back once we have a chance to work on Python dependency management a bit more and provide clear documentation on how to manage these things.

In the interim, if you are using pyenv to manage your local Python installations, please be sure you have a global version set, e.g., pyenv global 3.5.4.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+          name: Run tests
+          command: |
+            export NVM_DIR="$HOME/.nvm"
+            [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+            [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
+            . venv/bin/activate
+            nvm use default
+            cd fec
+            ./manage.py test
+            npm run test-single
```
Comment (ccostino) [inline .circleci/config.yml]: @xtine should this kick back up one directory before running the tests with `npm`?

Comment (xtine) [reply .circleci/config.yml]: npm tasks still run without being in the root directory.

**Comment by ccostino** [comment]
We're currently running into an issue where the locales available in the Python Docker image (and this is an issue in both the official Python Docker image as well as the CircleCI version) don't support currency formatting, which is used in one of our tests:

```
======================================================================
ERROR: test_currency_filter_not_none (data.tests.test_utils.TestUtils)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/repo/fec/data/tests/test_utils.py", line 15, in test_currency_filter_not_none
    assert filters.currency(1.05) == '$1.05'
  File "/home/circleci/repo/fec/data/templatetags/filters.py", line 21, in currency
    return locale.currency(num, grouping=grouping)
  File "/usr/local/lib/python3.5/locale.py", line 262, in currency
    raise ValueError("Currency formatting is not possible using "
ValueError: Currency formatting is not possible using the 'C' locale.

----------------------------------------------------------------------
```

We're trying to figure out ways to solve this because unfortunately we cannot just switch locales to one that will work.  The only available ones are `C`, `C.UTF-8`, and `POSIX`, which don't support currency formatting.

**Comment by ccostino** [comment]
I've commented out the failing test for now, I think it's safe to just get this working and then... circle back on fixing that (sorry!).

Now I'm stuck on a Django `SECRET_KEY` setting issue, but this is in the deployment step at this point at least!

**Comment by ccostino** [comment]
Thank you @LindsayYoung for the help with that!

**Comment by ccostino** [comment]
Hmm, guess we should close this one in favor of PR #1270?

**Pull request closed without merging**